### ISSUE TITLE
CATEGORIES fields destruction and other concerns about format

### ISSUE THREAD

**Issue by tkapias** [pimutils/todoman#491]
Hi,

First of all, thank you for creating Todoman, otherwise I would not have switches 100% to cli for mails, calendars, contacts and todos.

But, I have several issues with Todoman, especially with the categories and the destruction of the original data.

My todo stack:
- Desktop:
  - vdirsyncer (0.18.0 from Debian repo)
  - todoman (4.1.0 from Debian repo)
- Server:
  - [Baikal](https://github.com/sabre-io/Baikal)/SabreDAV (4.3.1)
- Android:
  - DAVx5 (4.2.1 from Fdroid)
  - [Tasks.org](https://github.com/tasks/tasks) (12.6.1 from Fdroid)

**Despite what other reports suggest, I observe that categories are not yet supported on Todoman. But worse, properly formatted data is ruined by Todoman.**

- The --category argument is still not working.
- The categories are not displayed, either in list, show or edit.
- If the ics file contains more than one category per line, **Todoman destroys the information**.
- If there is only one category, sometimes it leaves it alone, **but eventually destroys it**.

I tried with VTODOs created on Tasks.org or on Evolution then synchronized by DAVx5->Baikal->Vdirsyncer:
- The created field is like this: CATEGORIES:cat1,cat2.
- **The CATEGORIES field still exists after synchronization between the 3 machines**.
- **The CATEGORIES field disappears after launching Todoman**.

In addition:
- Todoman adds VALUE=DATE: everywhere unlike other tools. (not a serious problem though)
- Todoman sometimes deletes DTSTART fields if they are added later externally. ( usually there is no reason to do this)
- Todoman doesn't see VTODOs that don't have a DUE date. ( regrettable for some use cases )

**Comment by WhyNotHugo** [comment]
For any issue, I need a sample `ics` file and a series of steps that you take to reproduce the issue. This is necessary to understand what happens and evaluate how to fix it.

> The --category argument is still not working.

What do you mean by "not working"? Some tests run `todo list --category...` and those seem to be working fine.

> The categories are not displayed, either in list, show or edit.

Correct, see: https://github.com/pimutils/todoman/issues/10

> If the ics file contains more than one category per line, Todoman destroys the information.

What exactly did you do? I'd like to be able to reproduce this, sounds like a serious bug.

> If there is only one category, sometimes it leaves it alone, but eventually destroys it.

Ditto.

> The CATEGORIES field disappears after launching Todoman.

`todoman` **does not** write to files when reading them. Did you edit the tasks? What exactly did you do? Editing a few with categories does not

**Comment by Schubisu** [comment]
I think it's correct that category support does not work in version 4.1.0. When I understand correctly from [here](https://github.com/pimutils/todoman/tags) and [here](https://pypi.org/project/todoman/#history), the releases (from December 2021) have been generated before category support was added (commit messages `feat(categories)`) in January 2022 (tui support in July 2022).

I can confirm that it works well when installed from latest github main branch. But the new release v4.2.0 as mentioned in the change log seems to be not yet available.

**Comment by WhyNotHugo** [comment]
Indeed, category support is unreleased. I guess there's nothing blocking a 4.2.0 release at this point.

**Comment by Schubisu** [comment]
Thanks, just installed 4.2.0 via pip :)

**Comment by tionis** [comment]
Can this be closed? I tested category support as mentioned in #10, and it seems to work quite fine in main.
